src/napari/_qt/qt_main_window.py

```diff
@@ -1303,17 +1303,32 @@ def _dock_widgets(self) -> MutableMapping[str, QtViewerDockWidget]:
         # other widget we should keep this name for a longer period
         warnings.warn(
             'The `_dock_widgets` property is private and should not be used in any plugin code. '
-            'Please use the `dock_widgets` property instead.',
+            'Use the `dock_widgets` property instead. If you need the dock wrapper, get it '
+            'via `dock_widgets[name].parent()` (or `.native.parent()` for magicgui widgets).',
```

Comment:
its good as-is, but here's a suggestion to be a little more clear.
```suggestion
            'To return the inner widget, use the `dock_widgets` property instead.'
            'If you need the dock wrapper, return it via `dock_widgets[name].parent()`'
            '(or `dock_widgets[name].native.parent()` for magicgui widgets).',
```